exonum-java-binding/core/src/main/java/com/exonum/binding/core/storage/indices/ProofMapIndexProxy.java

```diff
@@ -184,10 +254,11 @@ public boolean containsKey(K key) {
   /**
    * {@inheritDoc}
    *
-   * @param key a proof map key, must be 32-byte long when serialized
+   * @param key a proof map key
    * @param value a storage value to associate with the key
    * @throws IllegalStateException if this map is not valid
-   * @throws IllegalArgumentException if the size of the key is not 32 bytes
+   * @throws IllegalArgumentException if the size of the key is not 32 bytes (in case of a
+   *     no key hashing proof map)
```

Comment:
Here and elsewhere 'a no key hashing proof map' does not sound correct.
 * ... a non-key-hashing proof map
 * ... a proof map that does not hash keys
 * if the size of the key is not 32 bytes *and* this map uses non-hashed keys ...

Reply:
@Maria-Nosyk Would you tell us please which works best?

Reply:
The third variant is absolutely perfect!

Reply:
Brilliant, thanks!

Reply:
Is `non-key-hashing proof map` a bad option though? Third variant is a bit too wordy in some cases.

Reply:
Yup, the construction you suggest is, first, hard to understand, second, contradicts general requirements to technical documentation. The second variant suggested by @dmitry-timofeev is also acceptable.